Source/Urho3D/Audio/SoundSource.cpp

```diff
@@ -96,6 +99,8 @@ namespace Urho3D
 
 static const int STREAM_SAFETY_SAMPLES = 4;
 
+static const int VOLUME_DENOM = 4096;
```

Comment:
Just curious how did you decide on this magic number. It is 2^12 for 16bit and reduce to 2^4 for 8bit audio. The existing code for 16bit multiply the volume with 256.f and then divide it back later, but for 8bit leave the multiplied volume as it is. The proposed code changed that to 4096.f for 16bit. It does not change much for 16bit audio, but it looks quite different now for 8bit audio. It becomes mult and divide by 16.

Reply:
These mixer routines are using integer arithmetic. In the original code the [0.f..1.f] gain values scaled to [0..256] integer values, so there was an integer division by 256. I just changed this scale value to 4096 for better precision.
In the 8bit case there is another multiply by 256 so the volume level does not change,
and it is "built in" the denominator.
I tried to choose this "magic number" big enugh to lower the quantization effect, but not too big to avoid overflow and long lerping time.